src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -94,13 +89,11 @@ public override void Fail(string message, string detailMessage)
             catch
             {
                 stackTrace = "";
-            }
-            // Tracked by #32955: WriteAssert should write "stackTrace" rather than string.Empty. 
-            WriteAssert(string.Empty, message, detailMessage);
```

Comment:
Confirmed that in .NET Framework for both Trace and Debug, Fail(message) will print the stack trace.